File: AtCoder/ABC/182/python/C_To_3.py
N = input()


M = len(N)
res = M
for i in range(1, 1 << M):
    cnt = 0
    t = ""
    for j in range(M - 1, -1, -1):
        if i >> j & 1:
            cnt += 1
            t += N[j]
    if int(t) % 3 == 0:
        res = min(res, M - cnt)
if res == M:
    res = -1
print(res)

File: AtCoder/ABC/182/cpp/C_To_3.cpp
#include <iostream>
#include <stdio.h>
#include <string>

using namespace std;

string N;

int main() {
  cin >> N;
  int M = N.size();
  int res = M;
  for (int i = 1; i < 1 << M; ++i) {
    string t;
    int cnt = 0;
    for (int j = M - 1; j >= 0; --j) {
      if (i >> j & 1) {
        t += N[j];
        ++cnt;
      }
    }
    if (stoll(t) % 3 == 0) {
      res = min(res, M - cnt);
    }
  }
  if (res == M) {
    res = -1;
  }
  printf("%d\n", res);
}